debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/YugabyteDBConnector.java

```diff
@@ -61,10 +62,70 @@ public void start(Map<String, String> props) {
 
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
+        if (props == null) {
+            return Collections.emptyList();
+        }
+
+        if (props.containsKey("snapshot.mode") && props.get("snapshot.mode").equalsIgnoreCase("parallel")) {
+            LOGGER.info("Initialising parallel snapshot consumption");
+
+            final String tableIncludeList = props.get("table.include.list");
+            // Perform basic validations.
+            if (tableIncludeList == null) {
+                throw new DebeziumException("No table provided, provide a table in the table.include.list");
+            } else if (tableIncludeList.contains(",")) {
+                // This might indicate the presence of multiple tables in the include list, we do not want that.
+                throw new DebeziumException("parallel snapshot consumption is only supported with one table at a time");
+            }
+
+            // Publication auto create mode should now be for all tables.
+            if (props.containsKey("publication.autocreate.mode") && props.get("publication.autocreate.mode").equalsIgnoreCase("all_tables")) {
+                throw new DebeziumException("Snapshot mode parallel is not supported with publication.autocreate.mode all_tables, " +
+                                            "use publication.autocreate.mode=filtered");
+            }
+
+            // Add configuration for select override.
+            props.put("snapshot.select.statement.overrides", props.get("table.include.list"));
+
+            return getConfigForParallelSnapshotConsumption(maxTasks);
+        }
+
         // this will always have just one task with the given list of properties
         return props == null ? Collections.emptyList() : Collections.singletonList(new HashMap<>(props));
     }
 
+    protected List<Map<String, String>> getConfigForParallelSnapshotConsumption(int maxTasks) {
+        List<Map<String, String>> taskConfigs = new ArrayList<>();
+
+        final long rangeSize = (64 * 1024) / maxTasks;
```

Comment:
Wont we miss the hash range when maxTasks is not a multiple of (64*1024)? 
We want to cover the range 0 - 65535. Suppose we have 3 tasks:
Task-1: hash range = 0 - 21844
Task-2: hash range = 21845 - 43689
Task-3: hash range = 43690 - 65534